iotdb-core/datanode/src/main/java/org/apache/iotdb/db/pipe/event/common/tablet/PipeInsertNodeTabletInsertionEvent.java

```diff
@@ -130,22 +144,52 @@ public ProgressIndex getProgressIndex() {
 
   @Override
   public PipeInsertNodeTabletInsertionEvent shallowCopySelfAndBindPipeTaskMetaForProgressReport(
-      String pipeName, PipeTaskMeta pipeTaskMeta, String pattern) {
+      String pipeName, PipeTaskMeta pipeTaskMeta, String pattern, long startTime, long endTime) {
     return new PipeInsertNodeTabletInsertionEvent(
         walEntryHandler,
         progressIndex,
         isAligned,
         isGeneratedByPipe,
         pipeName,
         pipeTaskMeta,
-        pattern);
+        pattern,
+        startTime,
+        endTime);
   }
 
   @Override
   public boolean isGeneratedByPipe() {
     return isGeneratedByPipe;
   }
 
+  @Override
+  public boolean isEventTimeOverlappedWithTimeRange() {
+    try {
+      InsertNode insertNode = getInsertNode();
+      if (insertNode instanceof InsertRowNode) {
+        long timestamp = ((InsertRowNode) insertNode).getTime();
+        return startTime <= timestamp && timestamp <= endTime;
+      } else if (insertNode instanceof InsertTabletNode) {
+        long maxTimestamp = Long.MIN_VALUE;
+        long minTimestamp = Long.MAX_VALUE;
```

Comment:
```suggestion
        long maxTimestamp = ((InsertTabletNode) insertNode).getTimes()[0];
        long minTimestamp = ((InsertTabletNode) insertNode).getTimes()[((InsertTabletNode) insertNode).getTimes().length - 1];
```

Eements in times are sorted.